sshuttle/ssh.py

```diff
@@ -63,8 +62,32 @@ def empackage(z, name, data=None):
 
 def connect(ssh_cmd, rhostport, python, stderr, options):
     portl = []
+    password = False
+    ipv6 = False
+    ipv4 = False
+
+
+
+
+    if re.sub(r'.*@', '', rhostport or '').count(':') >= 1:
+        ipv6 = True
+
+        # fix For ipv6
+        l = (rhostport or '')
+        if len(l.split(':')) == 4 or len(l.split(':')) == 5:
+            l = (rhostport or '').rsplit('@', 1)
+            if len(l[0].split(":")) == 3:
+                portl = ['-p', str(int(l[0].split(":")[2]))]
+                password = l[0].split(":")[1]
+
+            if len(l[0].split(":")) == 2:
+                password = l[0].split(":")[1]
+
+            if len(l[0].split(":")) == 1:
+                portl = ['-p', str(int(l[1].split(':')[2]))]
+
+            username = l[0].split(":")[0]
```

Comment:
All of this seems to be getting very complicated. I think we need to separate parsing into another function. For example, maybe something like:

```
def parse_hostport(rhostport):
   ... deal with both ipv4 and ipv6 ...
   return username, password, host, port


username, password, host, port = parse_hostport(rhostport)

rhost = "{}@{}".format(username, host)
```

Reply:
just rerused old method